PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/springcloud/dependency/action/AddAzureDependencyAction.java

```diff
@@ -79,87 +81,118 @@ public boolean onActionPerformed(@NotNull AnActionEvent event, @Nullable Operati
+                        PluginUtil.showInfoNotificationProject(project, "Your project is update-to-date.",
+                                "No updates are needed.");
+                        return;
+                    }
+                    progressIndicator.setText("Applying versions ...");
+                    File pomFile = new File(mavenProject.getFile().getCanonicalPath());
+                    ProgressManager.checkCanceled();
+                    versionChanges.stream().filter(change -> managerDependencyVersionsMaps.containsKey(change.getKey())).forEach(change -> {
+                        String managementVersion = managerDependencyVersionsMaps.get(change.getKey()).getCurrentVersion();
+                        if (StringUtils.equals(change.getCompatibleVersion(), managementVersion)
+                                || SpringCloudDependencyManager.isCompatibleVersion(managementVersion, springBootVer)) {
+                            change.setCompatibleVersion("");
+                            change.setManagementVersion(managementVersion);
+                        }
+                    });
+                    if (!dependencyManager.update(pomFile, versionChanges)) {
+                        PluginUtil.showInfoNotificationProject(project, "Your project is update-to-date.",
+                                "No updates are needed.");
+                        return;
+                    }
+
+                    final VirtualFile vf = LocalFileSystem.getInstance().refreshAndFindFileByIoFile(pomFile);
+                    RefreshQueue.getInstance().refresh(true, false, null, new VirtualFile[]{vf});
+                    ApplicationManager.getApplication().invokeLater(() -> {
+                        FileEditorManager.getInstance(project).closeFile(vf);
+                        FileEditorManager.getInstance(project).openFile(vf, true, true);
+                        if (versionChanges.stream().anyMatch(t -> StringUtils.isNotEmpty(t.getCurrentVersion()))) {
+                            PluginUtil.showInfoNotificationProject(project,
+                                    "Azure Spring Cloud dependencies are updated successfully.",
+                                    summaryVersionChanges(versionChanges));
+                        } else {
+                            PluginUtil.showInfoNotificationProject(project,
+                                    "Azure Spring Cloud dependencies are added to your project successfully.",
+                                    summaryVersionChanges(versionChanges));
+                        }
+                    });
+                } catch (DocumentException | IOException | AzureExecutionException | MavenProcessCanceledException e) {
+                    PluginUtil.showErrorNotification("Error",
+                            "Failed to update Azure Spring Cloud dependencies due to error: "
+                                    + e.getMessage());
+                }
+            });
```

Comment:
can this method be made simpler by extracting some well-named methods.

Reply:
done